### ISSUE TITLE
config.yml written without comments on Spigot

### ISSUE THREAD

**Issue by SlimeDog** [EngineHub/WorldGuard#2078]
### WorldEdit Version

7.3.1-6745

### WorldGuard Version

7.0.10-2266

### Platform Version

Spigot 1.20.6-4142

### Confirmations

- [X] I am using the most recent Minecraft release.
- [X] I am using a version of WorldEdit compatible with my Minecraft version.
- [X] I am using a version of WorldGuard compatible with my Minecraft version.
- [X] I am using the latest or recommended version of my platform software.
- [X] I am NOT using a hybrid server, e.g. a server that combines Bukkit and Forge. Examples include Arclight, Mohist, and Cardboard.
- [X] I am NOT using a fork of WorldEdit, such as FastAsyncWorldEdit (FAWE) or AsyncWorldEdit (AWE)

### Bug Description

On server start-up, `plugins/WorldGuard/config.yml` is (re)written without comments at top of file. The comments are contained in the `defaults/config.yml` file in the JAR. The comments are (re)written on Paper 1.20.6, and on both Paper and Spigot with previous versions of WorldGuard.

### Expected Behavior

Rewrite file with comments, regardless of platform.

### Reproduction Steps

1. Update to Spigot 1.20.6
2. Update to WorldGuard build 2266
3. Start server
4. Compare files

**Comment by wizjany** [comment]
i don't think we do anything paper-specific here tbh, only way i can see this happening is if spigot ships a broken snakeyaml lib and paper doesn't somehow.

**Comment by SlimeDog** [comment]
OK, thanks. I'll pursue that line.

**Comment by SlimeDog** [comment]
```
$ unzip -l paper-1.20.6-36.jar | grep snakeyaml
        0  05-02-2024 11:29   META-INF/libraries/org/yaml/snakeyaml/
        0  05-02-2024 11:29   META-INF/libraries/org/yaml/snakeyaml/2.2/
   334352  05-02-2024 11:29   META-INF/libraries/org/yaml/snakeyaml/2.2/snakeyaml-2.2.jar
```
```
$ unzip -l spigot-1.20.6-4142.jar | grep snakeyaml
   334352  05-03-2024 09:06   META-INF/libraries/snakeyaml-2.2.jar
```

**Comment by SlimeDog** [comment]
In addition, on Paper, WorldGuard created the following on server start-up
```
$ echo paper-*
paper-1.20.6-36.jar
$ tree migrations/
migrations/
└── region
    ├── mysql
    │   ├── V1__Initial.sql
    │   └── V2__Bug_fix_and_UUID.sql
    └── sqlite
        └── V1__Initial.sql
```
but not on Spigot (or perhaps the subdirectory was created, but then removed).

**Comment by aurorasmiles** [comment]
This doesn't seem to be a spigot-specific issue, I get the same behaviour on Paper. When the files are initially created they have comments, after you restart your server once the comments are gone.

**Comment by SlimeDog** [comment]
> This doesn't seem to be a spigot-specific issue, I get the same behaviour on Paper. When the files are initially created they have comments, after you restart your server once the comments are gone.

Agreed. Paper wasn't available when I originally posted the issue. Changed title to reflect generality.
